great_expectations/dataset/base.py

```diff
@@ -3344,8 +3355,473 @@ def expect_column_pair_values_to_be_in_set(self,
 
         """
         raise NotImplementedError
+        
+        
+    
+class DataFile(Dataset):
+    @classmethod
+    def file_map_expectation(cls, func):
+        
+        """Constructs an expectation using file-map semantics.
 
+        The file_map_expectations decorator handles boilerplate issues surrounding the common pattern of \
+        evaluating truthiness of some condition on an line by line basis in a file.
+
+        Args:
+            func (function): \
+                The function implementing an expectation that will be applied line by line across a file \
+                The function should take a file and return information about how many lines met expectations.
+
+        Notes:
+            Users can specify skip value k that will cause the expectation function to disregard the first
+            k lines of the file
+
+        See also:
+            :func:`expect_file_line_regex_match_count_to_be_between <great_expectations.dataset.base.Dataset.expect_file_line_regex_match_count_to_be_between>` \
+            for an example of a file_map_expectation
+        """
+        raise NotImplementedError
+        
+        
+        
+##### Output generation #####
+    def _format_file_map_output(self,
```

Comment:
This is duplicated boilerplate it looks like (or in the wrong spot)...are you intending for this to support mapping over multiple files in a dataset?

Reply:
Yes, this is duplicated directly from `format_column_map_output`; it only has the new name.  Both `DataTable` and `FileData` need this format output functionality (and they both need the `_calc_map_expectation_success` functionality). Right now, the `DataTable` and `FileDataset` objects have their own versions of essentially the same code. If you have no objections, I can move a general version of the `format_ output` and the `_calc_map_expectation_success` to the general `expectation` object from which both `DataTable` and `FileDataset` inherit. This would eliminate the duplication.

Yes, my choice of naming convention was confusing. It should be something like `file_line_map` not `file_map`. I will change all of that to make it more clear.